lib/annotated/tipset_tracking.go

```diff
@@ -0,0 +1,371 @@
+package annotated
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/jackc/pgx/v4"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin"
+	"github.com/multiformats/go-varint"
+)
+
+func (cs *acs) noteAccess(dbID uint64, t time.Time, atype accessType) {
+	cs.mu.Lock()
```

Comment:
@willscott actually you need to `return()` at the top here. Otherwise the `cs.accessStatsRecent` will keep growing without bounds: it is `SetCurrentTipset` that purges it